sharktank/sharktank/utils/iree.py

```diff
@@ -40,6 +40,47 @@
 }
 
 
+def get_iree_compiler_flags(
+    iree_hal_target_device: str,
+    iree_hal_local_target_device_backends: list[str] | None = None,
+    iree_hip_target: str | None = None,
+    device_count: int = 1,
+) -> list[str]:
+    """Retrieve compiler flags driven by the test configuration."""
+    res = []
+    if device_count == 1:
+        res += [f"--iree-hal-target-device={iree_hal_target_device}"]
+    else:
+        res += [
+            f"--iree-hal-target-device={iree_hal_target_device}[{i}]"
+            for i in range(device_count)
+        ]
+    if iree_hal_target_device.startswith("local"):
+        res += [
+            f"--iree-hal-local-target-device-backends={v}"
+            for v in iree_hal_local_target_device_backends
+        ]
+        res += ["--iree-llvmcpu-target-cpu=host"]
+    elif iree_hal_target_device.startswith("hip"):
+        res += [f"--iree-hip-target={iree_hip_target}"]
```

Comment:
Should raise an error on an else if they pass a different target deivce.

Reply:
Done.